server/src/socket/index.js

```diff
@@ -0,0 +1,52 @@
+import mongoose from 'mongoose';
+
+import Users from '../Schemas/users';
+import Messages from '../Schemas/messages';
+import Channels from '../Schemas/channels';
+import { io } from '../setup';
+const connectedUserData = {};
+io.on('connection', async client => {
+
+  client.on('userConnected', async ({ userName }) => {
+    const user = await Users.findOne({ username: userName });
+    const userData = { ...user._doc, socketId: client.id };
+    connectedUserData[userName] = { ...user, socketId: client.id };
+    client.emit('connectedUserAck', userData);
+  });
+
+  client.on('newMessage', async ({ channelId, message, username }) => {
+    
+    let channel = await Channels.findOne({ _id: channelId });
+    const participantsOfConversation = channel.users;
+    
+    if (participantsOfConversation.indexOf(username) < 0) {
+      return;
+    }
+    
+    const messageData = {
+      message,
+      channelId: mongoose.Types.ObjectId(channelId),
+      sender: username,
+      timestamp: new Date()
+    };
+    
+    new Messages(messageData).save();
```

Comment:
This is an async request i guess

Reply:
I did this because I didn't want the program to wait for the message to be saved.
But this might cause a problem if the DB operation fails.
Updating.

Reply:
Fixed